src/vector/transaction.rs

```diff
@@ -177,11 +177,48 @@ impl Dataset {
         }
         Ok(Transaction::new(self))
     }
+
+    /// Optionally start a transaction before running `func` for performance
+    ///
+    /// This uses transaction if the dataset supports it, otherwise it
+    /// runs the `func` function as it is on the dataset. If the
+    /// closure results in error, and it is a transaction, it is
+    /// rolled back. If there is error in rollback it is ignored.
```

Comment:
```suggestion
    /// rolled back. If the rollback fails, it not reported.
    ///
    /// <div class="warning">This function offers no ACID guarantees and
    /// no way to tell if the rollback succeeded or not. Do not use it if you
    /// care about your destination dataset.</div>
```

Reply:
I didn't check if this renders fine, please try `cargo doc --open` before amending.